File: server_integration_tests/servers/java_server_reactive/src/main/java/dev/graphlink/javaserver/impl/BulkCreateServiceImpl.java
package dev.graphlink.javaserver.impl;

import dev.graphlink.javaserver.generated.inputs.CreateArticleInput;
import dev.graphlink.javaserver.generated.services.BulkCreateService;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;

import java.util.List;

@Service
public class BulkCreateServiceImpl implements BulkCreateService {

    @Override
    public Mono<Integer> bulkCreate(List<? extends List<CreateArticleInput>> matrix) {
        int count = 0;
        return Mono.just(count);
    }

    @Override
    public Mono<Void> validateBulkCreate(List<? extends List<CreateArticleInput>> matrix) {
        return Mono.empty();
    }
}

File: server_integration_tests/servers/kotlin_server_blocking/src/main/kotlin/dev/graphlink/kotlinserverblocking/impl/BulkCreateServiceImpl.kt
package dev.graphlink.kotlinserverblocking.impl

import dev.graphlink.kotlinserverblocking.generated.inputs.CreateArticleInput
import dev.graphlink.kotlinserverblocking.generated.services.BulkCreateService
import org.springframework.stereotype.Service

@Service
class BulkCreateServiceImpl : BulkCreateService {
    override fun bulkCreate(matrix: List<List<CreateArticleInput>?>?): Int = 0

    override fun validateBulkCreate(matrix: List<List<CreateArticleInput>?>?) {}
}
